source/Ship.cpp

```diff
@@ -2640,7 +2642,17 @@ int Ship::TakeDamage(const Projectile &projectile, bool isBlast)
 	else if(shieldDamage > shields)
 		shieldFraction = min(shieldFraction, shields / shieldDamage);
 	shields -= shieldDamage * shieldFraction;
-	hull -= hullDamage * (1. - shieldFraction);
+	
+	double hullFraction = (1. - shieldFraction);
+	double disabledFraction = 0;
+	double nonDisabledHull = hull - MinimumHull() + .25d;
+	if(hullDamage * hullFraction > nonDisabledHull)
+	{
+		hullFraction = max(nonDisabledHull/hullDamage,0.d);
```

Comment:
Division by zero error if hullDamage is 0. (Can happen when nonDisabledHull is below zero on disabled target.)

Reply:
In the fixed version, nonDisabledHull is always positive so this won't happen, same case with a new variable, healPart.